plutus-scb/src/Cardano/Metadata/Server.hs

```diff
@@ -127,14 +129,19 @@ handleMetadata =
         GetProperty subject propertyKey -> do
             logInfo $ FetchingProperty subject propertyKey
             pure $ fetchById subject propertyKey
+        BatchQuery query@(QuerySubjects subjects) -> do
+            logInfo $ Querying query
+            pure . fromMaybe [] . traverse fetchSubject $ Set.toList subjects
 
 ------------------------------------------------------------
 handler ::
        Member MetadataEffect effs
-    => Subject
-    -> Eff effs (Maybe (SubjectProperties 'AesonEncoding))
-       :<|> (PropertyKey -> Eff effs (Maybe (Property 'AesonEncoding)))
-handler subject = getProperties subject :<|> getProperty subject
+    => (Subject -> (Eff effs (Maybe (SubjectProperties 'AesonEncoding))
+                    :<|> (PropertyKey -> Eff effs (Maybe (Property 'AesonEncoding)))))
+       :<|> (Query -> Eff effs [SubjectProperties 'AesonEncoding])
+handler =
+    (\subject -> getProperties subject :<|> getProperty subject) :<|>
+    batchQuery
```

Comment:
why isn't this `getProperties :<|> getProperty :<|> batchQuery`?

Reply:
See my previous comment.